keras/engine/saving.py

```diff
@@ -367,22 +373,25 @@ def save_model(model, filepath, overwrite=True, include_optimizer=True):
     if h5py is None:
         raise ImportError('`save_model` requires h5py.')
 
-    if not isinstance(filepath, h5py.Group):
-        # If file exists and should not be overwritten.
-        if not overwrite and os.path.isfile(filepath):
+    if H5Dict.is_supported_type(filepath):
+        if (
+            not overwrite and H5Dict.opens_file(filepath)
+            and os.path.isfile(filepath)
+        ):
             proceed = ask_to_proceed_with_overwrite(filepath)
             if not proceed:
                 return
-        opened_new_file = True
-    else:
-        opened_new_file = False
+        with H5Dict(filepath, mode='w') as h5dict:
+            _serialize_model(model, h5dict, include_optimizer)
+
```

Comment:
nit: I find whitespace before `else` or `elif` makes the code harder to read for me, but this is a personal preference